Error in SQLA (DetachedInstanceError) (in AiiDA v1.5 and newly reported in AiiDA v2.0.1)

### Describe the bug
Roughly half of my calculations are failing on a SQLA backend:
```
aiida.orm.nodes.process.calculation.calcjob.CalcJobNode: [ERROR] iteration 4 of do_update excepted, retrying after 160 seconds
Traceback (most recent call last):
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/engine/utils.py", line 172, in exponential_backoff_retry
    for iteration in range(max_attempts):
  File "/home/pizzi/.virtualenvs/aiida-prod/lib/python3.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/pizzi/.virtualenvs/aiida-prod/lib/python3.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "/home/pizzi/.virtualenvs/aiida-prod/lib/python3.7/site-packages/tornado/gen.py", line 307, in wrapper
    yielded = next(result)
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/engine/processes/calcjobs/tasks.py", line 184, in do_update
    with job_manager.request_job_info_update(authinfo, job_id) as update_request:
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/engine/processes/calcjobs/manager.py", line 279, in request_job_info_update
    with self.get_jobs_list(authinfo).request_job_info_update(job_id) as request:
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/engine/processes/calcjobs/manager.py", line 165, in request_job_info_update
    self._ensure_updating()
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/engine/processes/calcjobs/manager.py", line 188, in _ensure_updating
    self._update_handle = self._loop.call_later(self._get_next_update_delay(), updating)
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/engine/processes/calcjobs/manager.py", line 221, in _get_next_update_delay
    minimum_interval = self.get_minimum_update_interval()
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/engine/processes/calcjobs/manager.py", line 75, in get_minimum_update_interval
    return self._authinfo.computer.get_minimum_job_poll_interval()
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/orm/authinfos.py", line 81, in computer
    return computers.Computer.from_backend_entity(self._backend_entity.computer)
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/orm/implementation/sqlalchemy/authinfos.py", line 76, in computer
    return self.backend.computers.from_dbmodel(self._dbmodel.dbcomputer)
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/orm/implementation/sqlalchemy/utils.py", line 64, in __getattr__
    if self.is_saved() and self._is_mutable_model_field(item) and not self._in_transaction():
  File "/home/pizzi/.virtualenvs/aiida-prod/codes/aiida-core/aiida/orm/implementation/sqlalchemy/utils.py", line 87, in is_saved
    return self._model.id is not None
  File "/home/pizzi/.virtualenvs/aiida-prod/lib/python3.7/site-packages/sqlalchemy/orm/attributes.py", line 287, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/home/pizzi/.virtualenvs/aiida-prod/lib/python3.7/site-packages/sqlalchemy/orm/attributes.py", line 718, in get
    value = state._load_expired(state, passive)
  File "/home/pizzi/.virtualenvs/aiida-prod/lib/python3.7/site-packages/sqlalchemy/orm/state.py", line 652, in _load_expired
    self.manager.deferred_scalar_loader(self, toload)
  File "/home/pizzi/.virtualenvs/aiida-prod/lib/python3.7/site-packages/sqlalchemy/orm/loading.py", line 944, in load_scalar_attributes
    "attribute refresh operation cannot proceed" % (state_str(state))
sqlalchemy.orm.exc.DetachedInstanceError: Instance <DbAuthInfo at 0x7f1a6a57da10> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/13/bhk3)
```

I'm quite sure this was a production AiiDA 1.4.2 environment where I hadn't done anything and things were working fine until a few weeks ago.
Running yesterday, maybe calculations were pased with 5 consecutive errors and the error above.
I decided to stop the deamon, reinstall AiiDA 1.5.0 and replay them, but they fail again with the same error.

Any idea of what could be causing this?

@sphuber @CasperWA @chrisjsewell

--- a/aiida/engine/processes/calcjobs/manager.py
+++ b/aiida/engine/processes/calcjobs/manager.py
@@ -151,7 +151,7 @@ class JobsList:
             self._job_update_requests = {}
 
     @contextlib.contextmanager
-    def request_job_info_update(self, job_id: Hashable) -> Iterator['asyncio.Future[JobInfo]']:
+    def request_job_info_update(self, authinfo: AuthInfo, job_id: Hashable) -> Iterator['asyncio.Future[JobInfo]']:
         """Request job info about a job when the job next changes state.
 
         If the job is not found in the jobs list at the update, the future will resolve to `None`.
@@ -159,6 +159,7 @@ class JobsList:
         :param job_id: job identifier
         :return: future that will resolve to a `JobInfo` object when the job changes state
         """
+        self._authinfo = authinfo
         # Get or create the future
         request = self._job_update_requests.setdefault(job_id, asyncio.Future())
         assert not request.done(), 'Expected pending job info future, found in done state.'
@@ -283,7 +284,7 @@ class JobManager:
         This is a context manager so that if the user leaves the context the request is automatically cancelled.
 
         """
-        with self.get_jobs_list(authinfo).request_job_info_update(job_id) as request:
+        with self.get_jobs_list(authinfo).request_job_info_update(authinfo, job_id) as request:
             try:
                 yield request
             finally:

